android/sdl_android/src/androidTest/java/com/smartdevicelink/managers/SdlManagerTests.java

```diff
@@ -422,4 +422,20 @@ public void onResponse(int correlationId, RPCResponse response) {
         // Make sure the listener is called exactly once
         assertEquals("Listener was not called or called more/less frequently than expected", 1, listenerCalledCounter);
     }
+
+    @Test
+    public void testDisposeAfterRetryChangeRegistration() {
+        LockScreenConfig lockScreenConfig = new LockScreenConfig();
+        SdlManager sdlManager = createSampleManager("Test App", "test.test.test", lockScreenConfig);
+        sdlManager.retryChangeRegistration();
+        sdlManager.dispose();
+    }
+
+    @Test
+    public void testRetryChangeRegistrationAfterDispose() {
+        LockScreenConfig lockScreenConfig = new LockScreenConfig();
+        SdlManager sdlManager = createSampleManager("Test App", "test.test.test", lockScreenConfig);
+        sdlManager.dispose();
+        sdlManager.retryChangeRegistration();
+    }
```

Comment:
I dont think these unit tests actually will catch what we are trying to prevent. If the `checkLifecycleConfiguration` method is updated to manually throw an exception these tests will still pass. It does not look like the code is waiting for the runnable to complete so I believe the tests are completing before the runnable even executes.